File: leetcode/2997.使数组异或和等于 K 的最少操作次数.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        p = 0
        n = len(nums)
        for num in nums:
            p ^= num
        if p == k:
            return 0
        p = p ^ k
        ct = 0
        while p > 0:
            ct += p & 1
            p >>= 1
        return ct

File: leetcode/2997.使数组异或和等于 K 的最少操作次数.go
func minOperations(nums []int, k int) int {
	p := 0
	// n := len(nums)
	for _, num := range nums {
		p ^= num
	}
	if p == k {
		return 0
	}
	p = p ^ k
	ct := 0
	for p > 0 {
		ct += p & 1
		p >>= 1
	}
	return ct
}